wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetClientMessageRequestStrategy.swift

```diff
@@ -63,19 +63,18 @@ extension AssetClientMessageRequestStrategy: InsertedObjectSyncTranscoder {
         // Enter groups to enable waiting for message sending to complete in tests
         let groups = managedObjectContext.enterAllGroupsExceptSecondary()
         Task {
-            let result = await messageSender.sendMessage(message: object)
-
-            managedObjectContext.performAndWait {
-                switch result {
-                case .success:
+            do {
+                try await messageSender.sendMessage(message: object)
+                await managedObjectContext.perform {
                     object.markAsSent()
-                case .failure(let error):
+                }
+            } catch {
+                await managedObjectContext.perform {
                     object.expire()
 
-                    if case .networkError(let networkError) = error,
-                       case .invalidRequestError(let responseFailure, _) = networkError,
+                    if case NetworkError.invalidRequestError(let responseFailure, _) = error,
                        responseFailure.label == .missingLegalholdConsent {
-                        managedObjectContext.zm_userInterface.performGroupedBlock {
+                        self.managedObjectContext.zm_userInterface.performGroupedBlock {
```

Comment:
Do we create a retain cycle of `managedObjectContext` here? Honestly I'm currently not sure right now, but it's still early without ☕️😁

Reply:
Not sure :) the block only lives until it completes so after that I suppose there shouldn't be any cycles.